.github/workflows/ci.yml

```diff
@@ -0,0 +1,58 @@
+name: build and run tests
+on: [pull_request]
+
+jobs:
+  linux-docker:
+    runs-on: ubuntu-20.04
+    # The GH default is 360 minutes (it's also the max as of Feb-2021). However we should fail sooner.
+    # The only reason to exceed this time is if a test hangs.
+    timeout-minutes: 120
+    strategy:
+      # Enabling fail-fast would kill all Dockers if one of them fails. We want maximum output.
+      fail-fast: false
+      matrix:
+        # For every distro we want to test here, add one key 'distro' with a descriptive name, and one key
+        # 'containerId' with the name of the container (i.e., what you want to docker-pull)
+        include:
+          - distro: 'Ubuntu 22.04'
+            containerId: 'mormj/gr4-runtime-docker:ubuntu-22.04'
+    name: ${{ matrix.distro }}
+    container:
+      image: ${{ matrix.containerId }}
+      volumes:
+        - build_data:/build
+      options: --cpus 2
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout Project
+      - name: Install picoscope
+        run: |
+          su -c "apt-get update"
+          su -c "apt-get --yes install wget gnupg2 apt-utils udev"
+          # https://www.picotech.com/downloads/linux
+          wget -qO - https://labs.picotech.com/Release.gpg.key | su -c "apt-key add -"
+          su -c "bash -c 'echo \"deb https://labs.picotech.com/rc/picoscope7/debian/ picoscope main\" >/etc/apt/sources.list.d/picoscope7.list'"
+          su -c "apt-get update"
+          su -c "mkdir -p /etc/udev/rules.d"
+          su -c "apt-get --yes install libps3000a libps4000a libps5000a libps6000a"
+      - name: Install root
+        run: |
+          su -c "apt-get update"
+          # https://root.cern/install/dependencies/
+          su -c "apt-get --yes install dpkg-dev cmake g++ gcc binutils libx11-dev libxpm-dev libxft-dev libxext-dev python libssl-dev"
```

Comment:
I didn't need any of these X libs in my dockerfile
g++/gcc - are those not installed by the gr docker image already?
cmake - this is a general requirement for the build (as meson is using it indirectly to find some dependencies), so it should be moved out of the "Install root" section I think

Reply:
If a dependency is already present, it doesn't hurt and this makes it easier to synchronize it with further root updates as each dependency specifies it's full set of dependencies in its own step.
I'll comment out the X dependencies, to keep it identical with the install docs.

The build dependency on cmake for the gr-digitizer build is delivered from the image, which ships all the dependencies that are needed to generally build OOTs.